### ISSUE TITLE
IAM database authentication fails when local port is set other than 3306 in SSH port forwarding.

### ISSUE THREAD

**Issue by okkez** [awslabs/aws-advanced-jdbc-wrapper#473]
### Describe the bug

IamAuthConnectionPlugin will generate a wrong token if set up `MYSQL_CONNECTION_STRING` with non-default port such as 23306.
The port 23306 come from SSH port forwarding like below.

```console
local_port=23306
remote_port=3306
ssh -fNL ${local_port}:${db_host}:${remote_port} ${login_user}@${gateway_host}
```

sample code:

```java
package software.amazon;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.Properties;
import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
import software.amazon.awssdk.auth.credentials.SystemPropertyCredentialsProvider;
import software.amazon.jdbc.AwsWrapperProperty;
import software.amazon.jdbc.PropertyDefinition;
import software.amazon.jdbc.authentication.AwsCredentialsManager;
import software.amazon.jdbc.plugin.IamAuthConnectionPlugin;

public class AwsIamAuthenticationMysqlExample {
  public static final String MYSQL_CONNECTION_STRING =
      "jdbc:aws-wrapper:mysql://127.0.0.1:23306";
  private static final String USERNAME = "iam-db-auth-user";

  public static void main(String[] args) throws SQLException {

    final Properties properties = new Properties();

    // Enable AWS IAM database authentication and configure driver property values
    properties.setProperty(PropertyDefinition.PLUGINS.name, "iam");
    properties.setProperty(PropertyDefinition.USER.name, USERNAME);
    properties.setProperty(IamAuthConnectionPlugin.IAM_HOST.name, "my-aurora-cluster.cluster-xxxxxxxx.ap-northeast-1.rds.amazonaws.com");
    properties.setProperty(IamAuthConnectionPlugin.IAM_DEFAULT_PORT.name, "3306");
    properties.setProperty(IamAuthConnectionPlugin.IAM_REGION.name, "ap-northeast-1");

    // Attempt a connection
    try (Connection conn = DriverManager.getConnection(MYSQL_CONNECTION_STRING, properties);
        Statement statement = conn.createStatement();
        ResultSet result = statement.executeQuery("select 1")) {

      System.out.println(Util.getResult(result));
    }
  }
}
```

IamAuthConnectionPlugin will generate token using following parameters:
- database host: `my-aurora-cluster.cluster-xxxxxxxx.ap-northeast-1.rds.amazonaws.com`
- database port: `23306` from `MYSQL_CONNECTION_STRING`
- database region: `ap-northeast-1`
- database user: `iam-db-auth-user`

### Expected Behavior

IamAuthConnectionPlugin will generate proper token.

### What plugins are used? What other connection properties were set?

iam

### Current Behavior

IamAuthConnectionPlugin generates wrong token and fails database authentication.

### Reproduction Steps

1. Clone this repository
2. Build examples/AWSDriverExample/src/main/java/software/amazon/AwsIamAuthenticationMysqlExample
3. Paste sample code the above to AwsIamAuthenticationMysqlExample.java
4. Run main
   - Set credentials as you like. (I use environment variables AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY)

### Possible Solution

Define new property such as IAM_PORT to overwrite port from `hostSpec`.

https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/9be3a6a49a81118c294052e62dd0fd64cac9e426/wrapper/src/main/java/software/amazon/jdbc/plugin/IamAuthConnectionPlugin.java#L111

### Additional Information/Context

I want to use aws-advances-jdbc-wrapper with DBeaver.
Because I found this sentence [here](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/9be3a6a49a81118c294052e62dd0fd64cac9e426/docs/development-guide/DevelopmentGuide.md?plain=1#L57)

But I cannot find documents about DBeaver configuration in this repository.

### The AWS Advanced JDBC Driver version used

2.1.2

### JDK version used

17

### Operating System and version

Debian GNU/Linux sid

**Comment by crystall-bitquill** [comment]
Hi @okkez,

Thanks for reaching out and raising this issue.

We'll take a look at this and keep you updated as we investigate.

Thank you for your patience!

**Comment by aaronchung-bitquill** [comment]
Hello @okkez 

We have just merged in a fix and it is available in our latest snapshot build. 

The bug here was that the `IAM_DEFAULT_PORT` property is supposed to override and take priority over the `hostspec` port, but it was not. With the fix, if you specify the `IAM_DEFAULT_PORT` value way the you did in your sample code, it will work now. 

Could you kindly checkout our [snapshot build ](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/UsingTheJdbcDriver.md#using-a-snapshot-of-the-driver)and let us know if the issue still persists? 

Thank you!

**Comment by okkez** [comment]
@aaronchung-bitquill Thank you for your reply.

I've confirmed the issue is fixed in #475 and the sample code works expected.

**Comment by aaronchung-bitquill** [comment]
Excellent! We'll be closing this issue. Please feel free reach out again with any other issues you encounter. :)
